henson-plugin/src/main/java/dart/henson/plugin/internal/GenerateHensonNavigatorTask.java

```diff
@@ -91,11 +92,11 @@ public File getHensonNavigatorSourceFile() {
 
   @TaskAction
   public void generateHensonNavigator() {
-    JavaCompile javaCompiler = (JavaCompile) variant.getJavaCompiler();
+    TaskProvider<JavaCompile> javaCompiler = variant.getJavaCompileProvider();
```

Comment:
This is indeed purely cosmetic because it has always been executed during the task execution phase and doesn't entail the eager creation of the compile task.